src/main/java/org/jenkinsci/plugins/mesos/OfferQueue.java

```diff
@@ -0,0 +1,121 @@
+package org.jenkinsci.plugins.mesos;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.mesos.Protos;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+/**
+ * This class acts as a buffer of Offers from Mesos.  By default it holds a maximum of 100 Offers.
+ */
+public class OfferQueue {
+    private final Logger logger = Logger.getLogger(JenkinsScheduler.class.getName());
+
+    private static final int DEFAULT_CAPACITY = 100;
+    private static final Duration DEFAULT_OFFER_WAIT = Duration.ofSeconds(30);
```

Comment:
Can you please make DEFAULT_OFFER_WAIT configurable and default it to a minimum time ? For ex: We have 3000 + Jenkins instances. Each Jenkins holding its offers for 30 seconds may leave many Jenkins instances starving ...